core/shared/src/main/scala/org/http4s/headers/Content-Transfer-Encoding.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.parse.Parser
+import cats.parse.Parser.ignoreCase
+import org.typelevel.ci._
+
+/** This header specifies the way content is mime encoded.
+  *
+  * @see [[https://www.ietf.org/rfc/rfc2045.txt]]
+  */
+sealed abstract class `Content-Transfer-Encoding`(val value: CIString)
+    extends Product
+    with Serializable
```

Comment:
My reasoning was, that according to RFC there is a fixed list of allowed header values. Similar as `Cross-Origin-Resource-Policy` and then I modeled it similar to https://github.com/http4s/http4s/blob/series/0.23/core/shared/src/main/scala/org/http4s/headers/Cross-Origin-Resource-Policy.scala

Reply:
Maybe defining the primary constructor as `private` will be enough for that case, wdyt?

Reply:
How would one create/use a valid Header i.e here https://github.com/http4s/http4s/pull/7292/files#diff-7a8187454be930b09460cc0a66b14b3aa3d9534b7ad0eea37dbfc5c7be557519R74 ? 

Should I create a list of all valid vals in the companion object instead of the case objects?
```val binary = `Content-Transfer-Encoding`("binary")```

What would the benefits be? I don't want to argue against your idea just want to understand your reasoning :)

Reply:
lol, I've somehow misseen this part with case objects 🤦🏻 No objections to this solution then!

Reply:
No worries :)